src/metabase/search/core.clj

```diff
@@ -137,6 +136,17 @@
         (analytics/inc! :metabase-search/index-error)
         (throw e)))))
 
+(defn reindex!
+  "Populate a new index, and make it active. Simultaneously updates the current index.
+  Returns a future that will complete when the reindexing is done.
+  Respects `search.ingestion/*force-sync*` and waits for the future if it's true.
+  Alternately, if `:async?` is false, it will also run synchronously."
+  [& {:keys [async?] :or {async? true} :as opts}]
+  (let [f #(reindex-logic! opts)]
+    (if (or search.ingestion/*force-sync* (not async?))
+      (doto (promise) (deliver (f)))
+      (future (f)))))
```

Comment:
Just catching up - we went back to a future so that parallel tests can populate their temp indexes independently?

Reply:
"Back to the future" 🚗 ⚡